### PR TITLE
Firehose consumption using the Jetstream endpoint

### PR DESCRIPTION
**Pull request by levand**
Includes the implementation in `net.gosha.atproto.jetstream`.

Also includes a new `examples/jetstream_analysis.clj` file showing how to tap the firehose and collect some metrics on it. 

We did add a new dependency on the `charred` library since it's significantly more performant than `clojure.data.json`. This doesn't matter so much for normal API usage but for firehose-level volumes it can make a big difference. 

Implementation credit to @chase-lambert, I assisted some with design & review.

### CODE REVIEW

File: src/net/gosha/atproto/jetstream.clj [hunk 1]
```diff
@@ -0,0 +1,114 @@
+(ns net.gosha.atproto.jetstream
+  (:require
+   [charred.api           :as json]
+   [clojure.core.async    :as async]
+   [clojure.string :as string]
+   [clojure.tools.logging :as log])
+  (:import
+   [java.net URI]
+   [org.java_websocket.client WebSocketClient]))
```
Comment (goshatch) [inline src/net/gosha/atproto/jetstream.clj:9]: I was having a `ClassNotFoundException` exception in my REPL unless I added this to `deps.edn`:

```clojure
org.java-websocket/Java-WebSocket {:mvn/version "1.6.0"}
```

Do we need to depend on it explicitly, or is this some JDK config issue on my side?

File: src/net/gosha/atproto/jetstream.clj [hunk 2]
```diff
@@ -0,0 +1,114 @@
+(ns net.gosha.atproto.jetstream
+  (:require
+   [charred.api           :as json]
+   [clojure.core.async    :as async]
+   [clojure.string :as string]
```
Comment (goshatch) [inline src/net/gosha/atproto/jetstream.clj:5]: This require seems to be unused

**Review by goshatch — COMMENTED** [review]
Thank you so much, @chase-lambert and @levand! This is awesome.

Before merging, I was wondering if you could please:
- Add the `Java-Websocket` dependency if it's needed (see my comment)
- Update the README with a quick paragraph on how to get started with Jetstream
- If the implementation is fairly stable at this point, maybe some unit tests would be good?

Thank you again, I'm very looking forward to merging this!

**Comment by levand** [comment]
Will get these done, thanks! Except possibly the unit tests... this code doesn't really do anything at all except connect to an external service. By the time you mocked/stubbed enough to get a unit test, you'd basically not be testing anything at all.

It would certainly be valuable to think about what a functional or integration test suite might look like, though.

File: README.md [hunk 3]
```diff
@@ -34,7 +34,7 @@ The client is using [Martian](https://github.com/oliyh/martian/) under the hood
 (require '[net.gosha.atproto.core :as atproto]
          '[net.gosha.atproto.client :as atproto-client])
 
-(atproto/init {:baase-url "https://bsky.social"
+(atproto/init {:base-url "https://bsky.social"
```
Comment (goshatch) [inline README.md:37]: Nice catch, thank you!

**Comment by goshatch** [comment]
Thank you very much, @chase-lambert! I'm going to merge this.

**Pull request merged**